index.js

```diff
@@ -86,23 +86,23 @@ module.exports = class Hyperdrive extends ReadyResource {
     })
   }
 
-  flush () {
-    return this.files.flush()
+  async flush () {
+    await this.files.flush()
+    return this.close()
   }
 
   async _close () {
-    if (this._batching) return this.files.close()
     try {
```

Comment:
We should replace the try-catch for a `.catch(safetyCatch)`, much better because if we're already ignoring errors then at least we should ignore them individually per async call

I mean like this:
```js
const closeBlobs = this.blobs !== null && (this._checkout === null || this.blobs !== this._checkout.blobs)

if (closeBlobs) await this.blobs.core.close().catch(safetyCatch)
if (!this._batching) await this.db.close().catch(safetyCatch)
await this.files.close().catch(safetyCatch)
```
And if somehow we can make the `closeBlobs` thing clearer lol

Reply:
Benefit is that it allows to other things to try close as well, maybe not that much of a benefit but it's something

Reply:
Or just a try-catch to the whole function

Reply:
i think this is ok, and clear